File: 3471-minimum-average-of-smallest-and-largest-elements/minimum-average-of-smallest-and-largest-elements.cpp
class Solution {
public:
    double minimumAverage(vector<int>& nums) {
        vector<float>a;
        sort(nums.begin(),nums.end());
        int n = nums.size();
            for(int i=0; i<n/2; i++)
            {
                double avg = (float(nums[i]+nums[n-i-1]))/2;
                a.push_back(avg);
              
            }
       float minm = INT_MAX;
    for (float i : a) {
    minm = std::min(minm, i);
}
            return minm;
        
    }
};

File: 3471-minimum-average-of-smallest-and-largest-elements/minimum-average-of-smallest-and-largest-elements.java
class Solution {
    public double minimumAverage(int[] nums) {
       Arrays.sort(nums);
       int i =0 ;
       int j = nums.length-1;

       double min = Double.MAX_VALUE;

       while( i < j ){
        double average = (nums[i]+nums[j])/2.0;
        min = Math.min(min,average);
        i++;
        j--;
       }
       return min;

    }
}